src/Generator/Aggregator.fs

```diff
@@ -0,0 +1,88 @@
+﻿module Aggregator
+open Types
+open Util
+open SpecificationOpenGL
+open System
+
+let getEnumCasesAndCommandsPerVersion (data: OpenGL_Specification.Registry) =
+    let featuresSortedAscendingByVersion = 
+        data.Features
+        |> Array.sortBy(fun data -> data.Number)
+    let versions =
+        featuresSortedAscendingByVersion
+        |> Array.Parallel.map(fun data -> data.Name)
+    let resultsByVersion = Array.zeroCreate versions.Length
+    let aggregatedResultByVersion = Array.zeroCreate versions.Length
+    let getVersionBefore index = aggregatedResultByVersion.[index - 1]
+
+    let extractAddedAndRemoved(data: OpenGL_Specification.Feature) =
+        let (addedFunctions: string[], addedEnums : string[]) =
+            let a, b =
+                data.Requires
+                |> Array.Parallel.collectEither(fun e ->
+                    [| Left(e.Commands |> Array.Parallel.map(fun inner ->
+                        let res = inner.Name
```

Comment:
make this into a function to avoid duplicates

Reply:
Yes you got a point. Agreed.

Reply:
the workaround is ugly.